components/dashboard/src/start/StartWorkspace.tsx

```diff
@@ -75,8 +103,35 @@ export default class StartWorkspace extends React.Component<StartWorkspaceProps,
       this.setState({ error });
     }
 
+    // we're coming back from a workspace cluster/IDE URL where we expected a workspace to be running but it wasn't because either:
+    //  - this is a (very) old tab and the workspace already timed out
+    //  - due to a start error our workspace terminated very quickly between:
+    //    a) us being redirected to that IDEUrl (based on the first ws-manager update) and
+    //    b) our requests being validated by ws-proxy
+    // we break this potentially DDOS cycle by not calling "startWorkspace" automatically but offering a "Open workspace" button instead.
+    const params = this.props.parameters;
+    if (!!params?.not_found) {
+      this.setState({ dontAutostart: true });
+      this.fetchWorkspaceInfo(undefined);
+      return;
+    }
+
+    // IDE case:
+    //  - we assume the workspace has already been started for us
+    //  - we don't know it's instanceId
+    // we could also rely on "startWorkspace" to return us the "running" instance, but that opens the door for
+    // self-DDOSing as we found in the past (cmp. https://github.com/gitpod-io/gitpod/issues/8043)
+    if (this.runsInIFrame()) {
```

Comment:
The 'runsInIFrame` could also be asked outside of the component and then be passed as `autoStart: boolean` accordingly.